sdks/java/io/bigquery-io-perf-tests/src/test/java/org/apache/beam/sdk/bigqueryioperftests/BigQueryIOIT.java

```diff
@@ -101,6 +101,8 @@ public static void setup() throws IOException {
     metricsBigQueryTable = options.getMetricsBigQueryTable();
     testBigQueryDataset = options.getTestBigQueryDataset();
     testBigQueryTable = String.format("%s_%s", options.getTestBigQueryTable(), TEST_ID);
+    // BigQuery table names must contain only alphanumerics and underscores
+    testBigQueryTable = testBigQueryTable.replace("[^A-Za-z0-9_]+", "");
```

Comment:
I don't think this is the right solution to this. We hide some "magic" logic inside the test to avoid the error, whereas we should rather provide the proper name. 
I see that we generate the name in the jenkins JobDSL file and that is the place where we should solve the problem imo (of course if i'm not missing anything).

Reply:
Other than that I would avoid leaving a comment and assigning the variable twice. Rather than that, it would be better to have a well-named method encapsulating the behavior.

Reply:
I see that you posted new changes - the comments still apply :)

Reply:
The issue stems from the fact that in order to achieve unique table names (to avoid accidental collisions in case of multiple tests running at the same time) we append the test UUID to the table name. The UUID contains offending characters, not the name supplied in test definition.

I agree with the point about method extraction. I'll do just that. Thanks for the suggestion!